amp.php

```diff
@@ -157,3 +160,20 @@ function _amp_bootstrap_customizer() {
 	}
 }
 add_action( 'plugins_loaded', '_amp_bootstrap_customizer', 9 );
+
+/**
+ * Redirects the old AMP URL to the new AMP URL.
+ * If post slug is updated the amp page with old post slug will be redirected to the updated url.
+ *
+ * @param  string $link New URL of the post.
+ *
+ * @return string $link URL to be redirected.
+ */
+function amp_redirect_to_new_url( $link ) {
+
+	if ( is_amp_endpoint() ) {
+		$link = trailingslashit( $link ) . AMP_QUERY_VAR;
```

Comment:
Core ticket to add `$id` to the filter: https://core.trac.wordpress.org/ticket/39442